spec/interactors/mark_cart_as_processing_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+RSpec.describe MarkCartAsProcessing do
+  let(:interactor) {
+    MarkCartAsProcessing.new cart: cart, cart_version: cart_version
+  }
+  let(:context) { interactor.context }
+
+  let(:cart) { double }
+  let(:cart_version) { 3 }
+
+  context 'when the cart is processing' do
+    before do
+      allow(cart).to receive(:processing?).and_return(true)
+    end
+
+    before { interactor.call rescue Interactor::Failure }
+
+    it 'fails' do
+      expect(context).to_not be_a_success
+    end
+
+    it 'sets the correct error message' do
+      expect(context.message).to eq('generic.processing')
+    end
+  end
+
+  context 'when the cart is not processing' do
+    before do
+      allow(cart).to receive(:processing?).and_return(false)
+    end
+
+    context 'when the cart is fresh' do
+      before do
+        allow(cart).to receive(:update).with({
+          processing: true,
+          lock_version: cart_version
+        })
```

Comment:
Indent the right brace the same as the first position after the preceding left parenthesis.